### ISSUE TITLE
[Suggestion] Copy Paste Rotate

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#2886]
First got to say I love the copy paste. It is exactly how I think it should functionally perform. Colors and stuff can come later, but one issue is that it only rotates on the one axis you are pasting on. The normal blocks when rotating will rotate on all axis eventually. If we can have something like it for copy paste it would make life a bit easier.
![Rotate](https://user-images.githubusercontent.com/36980446/92204146-afe73a80-ee37-11ea-809e-e156a99ddca3.gif)

Additionally it could help if there was some sort of indicator where the center of the paste will be when you select a copy area.

**Comment by tsunamayo** [comment]
Yeah axis and  center are the two biggest offender in the usability.
For the axis I think I will add a way to switch the block axis, for the standard brick included.
For the center, yes I will add a preview. Also maybe a way to change the axis? if we go the prefab route.

**Comment by ZachZent** [comment]
R button for changing the axis? Not used for anything currently and close to the standard rotate buttons.

For the center, maybe just make the spot where you will be pasting have a yellow boundary and the rest white the size of a block

**Comment by tsunamayo** [comment]
R is the new standard button ;). Since Q and E are for roll in 6-dof.

**Comment by ZachZent** [comment]
Sweet

**Comment by BrickMacklin** [comment]
This was what I was referring to before with changing orientation

**Comment by Calvert4096** [comment]
I agree with ZachZent... the ability to rotate copied objects about any axis, and also *translate* relative to the surface under the cursor would be helpful.  Maybe using alt+WASD? As it is, the copy/paste tool is quite difficult to use without it clipping into the per-existing blocks.
